sherpa/astro/xspec/__init__.py

```diff
@@ -1791,6 +1836,10 @@ def guess(self, dep, *args, **kwargs):
         norm = get_xspec_norm(dep, self(*args))
         param_apply_limits(norm, self.norm, **kwargs)
 
+    @eval_xspec_with_fixed_norm
+    def calc(self, p, xlo, *args, **kwargs):
```

Comment:
The superclass - in this case `XSModel` has

```
def calc(self, p, *args, **kwargs):
```

so it might be best to keep this argument naming (our classes do not have a good history of that, but I'm trying to rain in the changes a little bit).

Reply:
It's a balance between consistency and readability.  In most parts of the models, we use 'p' for parameters (which are part of args here), while the superclass uses p for what all other routines call x0.

Reply:
I don't see one where the parent class uses `p` for `x0` - most of them use `p, *args, **kwargs` from what I can see. Some may use `pars` rather than `p`, and the "convolution" style models are an "all bets are off" case. I also am ignoring the optical / astro model classes as they are their own thing ....

Reply:
The point is that the first parameter is `xlo`. not `pars, but I'll change that for consistency.

Reply:
I still don't understand: we don't care what any argument is called after `p`/`pars` as we just send it through as `*args`:

```python
@superduperdecorator
def calc(self, p, *args, **kwargs):
    return super().calc(p, *args, **kwargs)
```

Or am I missing something important?

Reply:
No, you are all correct. It's just that the "p" in `calc(self, p, *args, **kwargs)` is not a parameter, it's x0, which is confusing to me. But I promise I'll just rename it and that's fine, too.

Reply:
`p` just seems like an usual way to abbreviate `x0`.

Reply:
Or maybe I'm confused. Either way, it works and I'll just change the name. I don't do anything with it at that location anyway.

Reply:
`p` should not be `x0` - the first argument for `calc` is always the parameter list. Now, the `__call__` version does ot take parameters, but that's a different routine/story.